### PR TITLE
ExternalId for Parameters

### PR DESCRIPTION
**Pull request by BerSie**
The term could also be applied to operation parameters. The annotation will then refer to a property of the operations binding parameter.

### CODE REVIEW

**Review by HeikoTheissen — COMMENTED** [review]
But a dynamic expression for a parameter can only be another parameter. Is that what you have in mind? Can you give an example?

**Comment by ralfhandl** [comment]
> a dynamic expression for a parameter can only be another parameter

I think for bound actions/functions a path expression could reach via the binding parameter into the binding target.

An example would be really helpful.

**Comment by BerSie** [comment]
> > a dynamic expression for a parameter can only be another parameter
> 
> I think for bound actions/functions a path expression could reach via the binding parameter into the binding target.

yes, exactly, that's the idea - I will check wrt an example.

**Review by HeikoTheissen — DISMISSED** [review]

**Comment by BerSie** [comment]
The example is there - is it correct and does it help? I create only an xml example - is the expectation to have an xml and a json example?

**Comment by BerSie** [comment]
indeed - cool - when I checked directly after pushing there was only xml (as created).

File: examples/Common.ExternalId-samples.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          <PropertyRef Name="PurchaseOrderId" />
+        </Key>
+        <Property Name="PurchaseOrderId" Type="Edm.Guid" Nullable="false" />
+        <Property Name="PurchaseOrder" Type="Edm.String" Nullable="true" MaxLength="10" />
+        <!-- ... and a lot more -->
+      </EntityType>
+
+      <Action Name="ChangePOReference" IsBound="true">
+        <Parameter Name="it" Type="this.SalesOrder_Type" />
+        <Parameter Name="PurchaseOrderId" Type="Edm.Guid" />
```
Comment (ralfhandl) [inline examples/Common.ExternalId-samples.xml:40]: Is this intentionally nullable to allow clearing the PO reference?

**Review by ralfhandl — CHANGES_REQUESTED** [review]

File: examples/Common.ExternalId-samples.xml (hunk 1 above)
Comment (BerSie) [reply examples/Common.ExternalId-samples.xml:40]: No - it was intentionally, but the possible reason you're mentioning sounds reasonable :-)

File: examples/Common.ExternalId-samples.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        <NavigationProperty Name="to_PurchaseOrder" Type="Collection(this.PurchaseOrder_Type)" />
+        <!-- ... and a lot more -->
+      </EntityType>
+
+      <EntityType Name="PurchaseOrder_Type">
+        <Key>
+          <PropertyRef Name="PurchaseOrderId" />
+        </Key>
+        <Property Name="PurchaseOrderId" Type="Edm.Guid" Nullable="false" />
+        <Property Name="PurchaseOrder" Type="Edm.String" Nullable="true" MaxLength="10" />
```
[the remark this answers is not in the document]
Comment (BerSie) [reply examples/Common.ExternalId-samples.xml:34]: No, it's the external (readable) id of the PurchaseOrderId. Maybe I should add this annotation as well (for better understanding)?

[the remark this answers is not in the document]
Comment (ralfhandl) [reply examples/Common.ExternalId-samples.xml:34]: Yes, please!

[the remark this answers is not in the document]
Comment (BerSie) [reply examples/Common.ExternalId-samples.xml:34]: Done

**Pull request merged**